### ISSUE TITLE
id prop is not actually required for Tabs

### ISSUE THREAD

**Issue by styu** [palantir/blueprint#4795]
#### Environment

- __Package version(s)__: 3.46.0
- __Operating System__: MacOS
- __Browser name and version__: 

#### Steps to reproduce

1. Use the `Tabs` component but leave out the `id` prop

#### Actual behavior

It compiles fine (screenshot from removing the following line:

https://github.com/palantir/blueprint/blob/5c05ce6bb12a9810b9ff35310a1554a65af6b65d/packages/docs-app/src/examples/core-examples/tabsExample.tsx#L69

![image](https://user-images.githubusercontent.com/1143755/124336501-3c53ae00-db6c-11eb-8ee5-f4c79179e739.png)

#### Expected behavior

The `id` prop should be required - this is necessary because the ID is used in `aria` attributes. Leaving it out leads to invalid attributes such that screen readers can't tell what the tabs are associated with

#### Possible solution

It seems that perhaps the default props is throwing off the typings?

https://github.com/palantir/blueprint/blob/5c05ce6bb12a9810b9ff35310a1554a65af6b65d/packages/core/src/components/tabs/tabs.tsx#L110
